Dockerfile.prod

```diff
@@ -0,0 +1,46 @@
+FROM node:10.14.1-alpine as build
+# working directory
+WORKDIR /usr/src/app
+
+# global environment setup : yarn + dependencies needed to support node-gyp
+RUN apk --no-cache --virtual add \
+    python \
+    make \
+    g++ \
+    yarn 
+
+# copy just the dependency files and configs needed for install
+COPY packages/peregrine/package.json ./packages/peregrine/package.json
+COPY packages/pwa-buildpack/package.json ./packages/pwa-buildpack/package.json
+COPY packages/upward-js/package.json ./packages/upward-js/package.json
+COPY packages/upward-spec/package.json ./packages/upward-spec/package.json
+COPY packages/venia-concept/package.json ./packages/venia-concept/package.json
+COPY package.json yarn.lock babel.config.js browserslist.js magento-compatibility.js ./
+
+# install dependencies with yarn
+RUN yarn install --frozen-lockfile
+
+# copy over the rest of the package files
+COPY packages ./packages
```

Comment:
Can you elaborate on why we can't just copy `packages` and run `yarn install`?

Reply:
Copying over just the files needed to install takes advantage of the docker cache, which allows for faster builds when there isn't a change to an image layer, each command in the dockerfile creates a new image layer as it builds. So if no changes are made to the dependencies then those layers will build from cache on subsequent builds. after hte install we copy over the rest of the files and if there are changes to those files then it will rebuild otherwise it will build from cache. More deets on the [docs](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#leverage-build-cache)